### PR TITLE
EFA Test Fixes

### PR DESCRIPTION
**Pull request by sean-smith**
Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/conftest.py [hunk 1]
```diff
@@ -167,7 +167,12 @@ def clusters_factory(request):
 
     def _cluster_factory(cluster_config):
         if request.config.getoption("cluster"):
-            cluster = Cluster(name=request.config.getoption("cluster"), ssh_key=request.config.getoption("key_path"))
+            cluster_config = _write_cluster_config_to_outdir(request, cluster_config)
```
Comment (demartinofra) [inline tests/integration-tests/conftest.py]: Minor: if and else branches differ only for the cluster name

File: tests/integration-tests/tests/test_efa/test_efa.py [hunk 2]
```diff
@@ -71,12 +72,14 @@ def _test_efa_mpi(remote_command_executor, scheduler_commands, test_datadir, slo
     )
 
     # submit script using additional files
-    result = scheduler_commands.submit_script(str(test_datadir / "mpi_submit.sh"), slots=2 * slots_per_instance)
+    result = scheduler_commands.submit_script(
+        str(test_datadir / "mpi_submit.sh"), slots=2, filename="mpi_submit.sh.o%j"
```
Comment (demartinofra) [inline tests/integration-tests/tests/test_efa/test_efa.py]: filename was defined in submit_command for slurm and submit_script for sge 

Also Filename is not used in sge implementation. If it’s easier to have cross schedulers compatibility you can leave the redirection on the mpirun command.

File: tests/integration-tests/tests/test_efa/test_efa.py [hunk 3]
```diff
@@ -86,10 +89,14 @@ def _test_osu_benchmarks(remote_command_executor, scheduler_commands, test_datad
...
+        str(test_datadir / "osu_submit.sh"), slots=2 * slots_per_instance, filename="osu_submit.sh.o%j"
+    )
     job_id = scheduler_commands.assert_job_submitted(result.stdout)
     scheduler_commands.wait_job_completed(job_id)
     scheduler_commands.assert_job_succeeded(job_id)
 
     # TODO: perform assertions on benchmarks results
-    remote_command_executor.run_remote_command("cat /shared/osu.out")
+    output = remote_command_executor.run_remote_command("cat osu_submit.sh.o{0}".format(job_id)).stdout
+    latency = re.search("0\s+(\d\d)\.", output).group(1)  # noqa: W605
```
Comment (demartinofra) [inline tests/integration-tests/tests/test_efa/test_efa.py]: Minor: have you tried r”0....” to fix the W605?

Also pls remove the todo above

File: tests/integration-tests/tests/test_efa/test_efa/test_efa/osu_submit.sh [hunk 4]
```diff
@@ -2,4 +2,4 @@
 set -e
 
 module load openmpi
-mpirun --map-by ppr:1:node /shared/osu-micro-benchmarks-5.4/mpi/pt2pt/osu_latency &> /shared/osu.out
+mpirun -N 1 -np 2 /shared/osu-micro-benchmarks-5.4/mpi/pt2pt/osu_latency
```
Comment (demartinofra) [inline tests/integration-tests/tests/test_efa/test_efa/test_efa/osu_submit.sh]: Did you try this with both sge and slurm?

File: tests/integration-tests/tests/test_efa/test_efa/test_efa/init_osu_benchmarks.sh [hunk 5]
```diff
@@ -7,4 +7,4 @@ tar zxvf ./osu-micro-benchmarks-5.4.tar.gz
 cd osu-micro-benchmarks-5.4/
 ./configure CC=/opt/amazon/efa/bin/mpicc CXX=/opt/amazon/efa/bin/mpicxx
 make
-# make install in the submit script
+sudo make install
```
Comment (demartinofra) [inline tests/integration-tests/tests/test_efa/test_efa/test_efa/init_osu_benchmarks.sh]: Minor: Is this needed?

File: tests/integration-tests/conftest.py (hunk 1 above)
Comment (sean-smith) [reply tests/integration-tests/conftest.py]: Changed 👍

File: tests/integration-tests/tests/test_efa/test_efa.py (hunk 2 above)
Comment (sean-smith) [reply tests/integration-tests/tests/test_efa/test_efa.py]: I've changed it to use `>>`, problem before was one mpi process output nothing, which cause the file to be empty. New behavior is each mpi process appends to the same file.

File: tests/integration-tests/tests/test_efa/test_efa.py (hunk 3 above)
Comment (sean-smith) [reply tests/integration-tests/tests/test_efa/test_efa.py]: changed 👌🏻

File: tests/integration-tests/tests/test_efa/test_efa/test_efa/init_osu_benchmarks.sh (hunk 5 above)
Comment (sean-smith) [reply tests/integration-tests/tests/test_efa/test_efa/test_efa/init_osu_benchmarks.sh]: sudo is needed

File: tests/integration-tests/tests/test_efa/test_efa/test_efa/osu_submit.sh (hunk 4 above)
Comment (sean-smith) [reply tests/integration-tests/tests/test_efa/test_efa/test_efa/osu_submit.sh]: nope

**Pull request merged**